### PR TITLE
Fix: Error boundaries and fetch errors handling

### LINKED ISSUE #285: Don't crash a page when a chart fails to render
**Describe the bug**
In the non-Lantern views, API requests are still handled at the page level instead of the chart level, meaning that if one chart fails to load in an inelegant way, the entire page will crash. This renders in production as a blank white page with errors in the console.

**To Reproduce**
Steps to reproduce the behavior by intentionally making a chart fail:
1. Go to the staging dashboard-data bucket
2. Download a file for safe-keeping, and then delete it from the bucket
3. Load the dashboard in staging or locally not in demo-mode
4. Navigate to the page with the chart backed by that metric file and see the entire page crash
5. Re-upload the deleted file to clean up

**Expected behavior**
Other charts should render and an error should appear in the place of the chart that failed to render.

**Additional context**
A greater piece of work is to migrate the non-Lantern views to the API request pattern of the Lantern views (#286), and have each chart request its own API call. That is a larger endeavor but might block this one.

### PR DESCRIPTION
**Pull request by Wmaileq**
## Description of the change

- Added fetching error catching
- Charts wrapped with ErrorBoundaries

## Type of change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

> Closes #285 

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Review by jessex — CHANGES_REQUESTED** [review]
Well done, @Wmaileq! Code looks solid to me. I don't think the error boundary is working exactly as prescribed, though, at least not in my testing. 

To test this, I updated the name of the file being requested in the API call from a given chart component, e.g. changing L68 of `RevocationMatrix.js` from `revocations_matrix_cells` to `bogusbogus`. You see that the error successfully renders in place of the chart, but then a few seconds later the page crashes anyway:

Screen Shot 2020-10-01 at 15 55 51

File: src/components/Error.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+const Error = ({ text }) => (
+  <div className="error">
+    <img src={warningIcon} alt="Error icon" className="error_icon" />
+    <p className="error_text">{text}</p>
+  </div>
+);
+
+Error.defaultProps = {
+  text: "Oops... Something went wrong. Check console for detailed info.",
```
[the remark this answers is not in the document]
Comment (sychang) [reply src/components/Error.js:16]: Copy suggestion: "Something went wrong while loading this chart. Check back later or contact feedback@recidiviz.org for more information."

**Comment by Wmaileq** [comment]
> Well done, @Wmaileq! Code looks solid to me. I don't think the error boundary is working exactly as prescribed, though, at least not in my testing. 
> 
> 
> 
> To test this, I updated the name of the file being requested in the API call from a given chart component, e.g. changing L68 of `RevocationMatrix.js` from `revocations_matrix_cells` to `bogusbogus`. You see that the error successfully renders in place of the chart, but then a few seconds later the page crashes anyway:
> 
> 
> 
> Screen Shot 2020-10-01 at 15 55 51
> 
> 

Yes, it is the dev environment issue, you can close error overlay by pressing 'esc' or cross at the right top of the screen or by disabling error overlays globally in dev. mode
In production it should work as expected.

**Review by jessex — APPROVED** [review]
We're good to go here! Was able to properly test by producing a build and running that locally against the dev database, changing one of the Lantern charts to use a bogus API path which doesn't exist in staging. Pushed up another commit which fixed some whitespace issues with the error message. Ready to merge! Thanks, @Wmaileq.

**Pull request merged**